icu4j/main/collate/src/main/java/com/ibm/icu/text/RuleBasedCollator.java

```diff
@@ -461,6 +461,7 @@ public void setLowerCaseFirst(boolean lowerfirst) {
      * @see #setUpperCaseFirst(boolean)
      * @stable ICU 2.8
      */
+    @SuppressWarnings("ReferenceEquality")
```

Comment:
FYI
These `if(settings.readOnly() == defaultSettings)` are deliberate and necessary.

Reply:
Yes, I figured, that's why I left them in place.